### PR TITLE
expose input variable name as variable

### PR DESCRIPTION
**Pull request by saig0**
Expose the input variable name of a decision table input as variable itself. This enables an expression / script engine to access the input variable, in case it's not the default variable 'cellInput'.

### CODE REVIEW

File: engine/src/main/java/org/camunda/bpm/dmn/engine/impl/evaluation/DecisionTableEvaluationHandler.java [hunk 1]
```diff
@@ -164,12 +163,16 @@ protected DmnEvaluatedDecisionRule evaluateMatchingRule(List<DmnDecisionTableOut
 
   protected VariableContext getLocalVariableContext(DmnDecisionTableInputImpl input, DmnEvaluatedInput evaluatedInput, VariableContext variableContext) {
     if (isNonEmptyExpression(input.getExpression())) {
+      String inputVariableName = evaluatedInput.getInputVariable();
+
       return CompositeVariableContext.compose(
-        SingleVariableContext.singleVariable(evaluatedInput.getInputVariable(), evaluatedInput.getValue()),
+        Variables.createVariables()
+            .putValueTyped(inputVariableName, evaluatedInput.getValue())
+            .putValue("inputVariableName", inputVariableName)
```
Comment (menski) [inline engine/src/main/java/org/camunda/bpm/dmn/engine/impl/evaluation/DecisionTableEvaluationHandler.java]: I would change the ordering of the putValue calls. So if for whatever reason an user decides to call its variable `inputVariableName` it will not be overridden by the second putValue. And would still work with the default implementations. I would also add a test for that case.

```java
Variables.createVariables()
  .putValue("inputVariableName", inputVariableName)
  .putValueTyped(inputVariableName, evaluatedInput.getValue())
```

File: engine/src/test/resources/org/camunda/bpm/dmn/engine/el/ExpressionEvaluationTest.variableContext.dmn [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+<definitions xmlns="http://www.omg.org/spec/DMN/20151101/dmn.xsd" xmlns:camunda="http://camunda.org/schema/1.0/dmn" id="definitions_0afw8ry" name="definitions" namespace="http://camunda.org/schema/1.0/dmn">
+  <decision id="decision" name="Decision">
+    <decisionTable id="decisionTable" hitPolicy="FIRST">
+      <input id="input1" label="in" camunda:inputVariable="foo">
+        <inputExpression id="inputExpression1" typeRef="integer">        <text>in</text>
+</inputExpression>
+      </input>
+      <output id="output1" label="out" name="out" typeRef="boolean" />
+      <rule id="row-914871961-1">
+        <inputEntry id="UnaryTests_1lhqhwf" expressionLanguage="juel">        <text><![CDATA[${variableContext.resolve('foo').getValue() == 3}]]></text>
```
Comment (menski) [inline engine/src/test/resources/org/camunda/bpm/dmn/engine/el/ExpressionEvaluationTest.variableContext.dmn]: Shouldn't this expression also evaluate the `inputVariableName` variable. Otherwise the test does not add anything as it also passes without your change.

```
${variableContext.resolve(variableContext.resolve('inputVariableName').getValue()).getValue() == 3}
```

Comment (saig0) [inline engine/src/test/resources/org/camunda/bpm/dmn/engine/el/ExpressionEvaluationTest.variableContext.dmn]: This test is not related to my changes. I just added it because there was no explicit test for the variable context.

**Comment by menski** [comment]
Thank you for your contribution :+1: merged with 20ef6572e18ced99e7bc3b1312d38283fed871e9
